gateway/managers/syncman/syncman_crud.go

```diff
@@ -557,69 +553,68 @@ func (s *Manager) GetSchemas(ctx context.Context, project, dbAlias, col, format
 		return http.StatusBadRequest, nil, err
 	}
 
-	if format == "json" {
-		a := s.modules.GetSchemaModuleForSyncMan()
-		if dbAlias != "*" && col != "*" {
-			collectionInfo, p := a.GetSchema(dbAlias, col)
-			if !p {
-				return http.StatusBadRequest, nil, helpers.Logger.LogError(helpers.GetRequestID(ctx), fmt.Sprintf("collection (%s) not present in config for dbAlias (%s) )", dbAlias, col), nil, nil)
-			}
-			fields := []*model.FieldType{}
-			for _, v := range collectionInfo {
-				fields = append(fields, v)
-			}
-			return http.StatusOK, []interface{}{map[string]*dbJSONSchemaResponse{fmt.Sprintf("%s-%s", dbAlias, col): {Fields: fields}}}, nil
-		} else if dbAlias != "*" {
-			collections := projectConfig.Modules.Crud[dbAlias].Collections
-			coll := map[string]*dbJSONSchemaResponse{}
-			for key := range collections {
-				collectionInfo, _ := a.GetSchema(dbAlias, key)
-				fields := []*model.FieldType{}
-				for _, v := range collectionInfo {
-					fields = append(fields, v)
-				}
-				coll[fmt.Sprintf("%s-%s", dbAlias, key)] = &dbJSONSchemaResponse{Fields: fields}
+	alreadyAddedTables := map[string]bool{}
+	schemaResponse := make([]DbSchemaResponse, 0)
+	a := s.modules.GetSchemaModuleForSyncMan()
+	if dbAlias != "*" && col != "*" {
+		if err := getSchemaResponse(ctx, format, dbAlias, col, true, alreadyAddedTables, a, projectConfig.Modules.Crud[dbAlias].Collections, &schemaResponse); err != nil {
```

Comment:
I think the schema module should have a function should return all collections in a db in order. Don't do the sorting in `syncman`. So have a function `schema.GetSchemaForDB(dbAlias)`.

Reply:
The test cases to test if things are sorted would be in the schema module. Not here.